src/security/core/src/dds_security_serialize.c

```diff
@@ -629,16 +632,17 @@ DDS_Security_Deserialize_PropertySeq(
        and just as good for checking that the length value isn't completely ridiculous. */
     const uint32_t minpropsize = (uint32_t) (2 * sizeof (uint32_t));
     int r = 1;
-    uint32_t i;
+    uint32_t length;
 
-    if (!DDS_Security_Deserialize_uint32_t(dser, &seq->_length)) {
+    if (!DDS_Security_Deserialize_uint32_t(dser, &length)) {
         return 0;
-    } else if (seq->_length > dser->remain / minpropsize) {
-        seq->_length = 0;
+    } else if (length > dser->remain / minpropsize) {
         return 0;
-    } else if (seq->_length > 0) {
+    } else if (length > 0) {
+        DDS_Security_PropertySeq_deinit(seq);
+        seq->_length = seq->_maximum = length;
```

Comment:
It is harmless because the first time through the sequence is correctly initialised to an empty one, and so it is only on further copies of the same property that it starts to matter. Data containing multiple copies of the same field but with different content is highly problematic but it really only affects consistency of interpretation between implementations. That in itself is manageable, but the inconsistencies can lead to all kinds of weird behaviours at the system level that are nearly impossible to debug. In my view these cases must therefore be rejected. That said, I don't want to deal with that in this bit of code: I'd much rather remove this deserialiser and instead reuse the one in `ddsi_plist.c` which already guards against that.

I did push a change because the behaviour for these two was inconsistent with the deserialisation code for other types: all others always return the latest data in the sample.

Good catch, thank you!